src/main/java/app/controller/ConfigController.java

```diff
@@ -69,6 +78,20 @@ public String getConfig(Model model) throws IOException {
     @PostMapping
     public String postConfig(@RequestParam final Map<String, Object> formParams) throws IOException {
         ConfigurationFileService.writeSettings(settingsPath, parseRootComponent.getRootComponent(formParams));
+              
+        // save data that is useful for other controllers to mongodb(db: application col: data)
+        Document applicationData= (new Document())
+        		.append("localNodePrivKey", formParams.get("privKeys"));
+    
+        final MongoCollection<Document> collection = mongoClient.getDatabase("application")
+        		.getCollection("data");
+                        
+        if(collection.countDocuments() == 0L) {        	
```

Comment:
Dont need that check. Use .save() instead:
Shorthand for insert/update is save - if _id value set, the record is updated if it exists or inserted if it does not;
if the _id value is not set, then the record is inserted as a new one.

https://mongodb.github.io/node-mongodb-native/markdown-docs/insert.html